test/spec/functions.ts

```diff
@@ -0,0 +1,184 @@
+}
+
+/**
+ * input:
+ *   pubkey: bytes48 -- the pubkey
+ *   message: bytes32 -- the message
+ *   signature: bytes96 -- the signature to verify against pubkey and message
+ * output: bool  -- VALID or INVALID
+ */
+function verify(input: {pubkey: string; message: string; signature: string}): boolean {
+  const {pubkey, message, signature} = input;
+  return VERIFY(fromHex(message), fromHex(pubkey), fromHex(signature));
+}
+
+/**
+ * ```
+ * input:
+ *   pubkeys: List[BLS Pubkey] -- the pubkeys
+ *   messages: List[bytes32] -- the messages
+ *   signature: BLS Signature -- the signature to verify against pubkeys and messages
+ * output: bool  --  true (VALID) or false (INVALID)
+ * ```
+ */
+function aggregate_verify(input: {pubkeys: string[]; messages: string[]; signature: string}): boolean {
+  const {pubkeys, messages, signature} = input;
+  return aggregateVerify(messages.map(fromHex), pubkeys.map(fromHex), fromHex(signature));
+}
+
+/**
+ * ```
+ * input:
+ *   pubkeys: List[BLS Pubkey] -- list of input BLS pubkeys
+ *   message: bytes32 -- the message
+ *   signature: BLS Signature -- the signature to verify against pubkeys and message
+ * output: bool  --  true (VALID) or false (INVALID)
+ * ```
+ */
+function fast_aggregate_verify(input: {pubkeys: string[]; message: string; signature: string}): boolean {
+  const {pubkeys, message, signature} = input;
+  return fastAggregateVerify(fromHex(message), pubkeys.map(fromHex), fromHex(signature));
+}
+
+/**
+ * ```
+ * input:
+ *   pubkeys: List[BLS Pubkey]
+ *   messages: List[Bytes32]
+ *   signatures: List[BLS Signature]
+ * output: bool -- true (VALID) or false (INVALID)
+ */
+function batch_verify(input: {pubkeys: string[]; messages: string[]; signatures: string[]}): boolean {
+  const length = input.pubkeys.length;
+  if (input.messages.length !== length && input.signatures.length !== length) {
+    throw new Error("Invalid spec test. Must have same number in each array. Check spec yaml file");
+  }
+  const sets: SignatureSet[] = [];
+  for (let i = 0; i < length; i++) {
+    sets.push({
+      message: fromHex(input.messages[i]),
+      publicKey: fromHex(input.pubkeys[i]),
+      signature: fromHex(input.signatures[i]),
+    });
+  }
+  return verifyMultipleAggregateSignatures(sets);
+}
+
+/**
+ * ```
+ * input:
+ *   pubkeys: List[BLS Pubkey] -- list of input BLS pubkeys
+ *   message: bytes32 -- the message
+ *   signature: BLS Signature -- the signature to verify against pubkeys and message
+ * output: bool  --  true (VALID) or false (INVALID)
+ * ```
+ */
+function eth_fast_aggregate_verify(input: {pubkeys: string[]; message: string; signature: string}): boolean {
+  const {pubkeys, message, signature} = input;
+
+  /**
+   * This is a bit of a hack. The altair spec changed this from being invalid to being valid.  Cannot
+   * accommodate both conditions from the base layer...
+   */
+  if (pubkeys.length === 0 && signature === G2_POINT_AT_INFINITY) {
+    return true;
+  }
```

Comment:
I really dont like this hack here because this was the reason this PR was created.  To remove all the code that was like this.  I am not sure how to remove this though.  The `bls` and `phase0` specs call for this condition to return `false` so that is what is returned from the native layer.  For `altair` this was changed to return `true`.  Without passing the fork, which this library has no view of, there is no way to get both false and true to return from identical test cases. 

So this hack had to stay....